preprocessing.py

```diff
@@ -1,115 +1,154 @@
 import pandas as pd
 import matplotlib.pyplot as plt
+import numpy as np
 from missingData import process_missing_DenverData2013,process_missing_AlamosaData2014
 
-fileNames = ['Alamosa-2003-2012.csv','Denver-2003-2012.csv','Grand-2003-2012.csv','Springs-2003-2012.csv']
-path = "C:/Users/ramosv/Desktop/GitHub/Weather-Prediction-Model/raw_data/Weather Prediction Raw Data/"
-allFrames = []
-
-for file in fileNames:
-    #os.path.join(path,file)...
-    temp_frame = pd.read_csv(path+file,usecols=["DATE","HourlyDryBulbTemperature"])
-    temp_frame["DATE"] = pd.to_datetime(temp_frame["DATE"])
-
-    temp_frame['HourlyDryBulbTemperature'] = pd.to_numeric(temp_frame['HourlyDryBulbTemperature'],errors='coerce')
-    temp_frame["HourlyDryBulbTemperature"].interpolate(inplace=True)
-    temp_frame["HourlyDryBulbTemperature"].fillna(method='ffill',inplace=True)
-    temp_frame["HourlyDryBulbTemperature"].fillna(method='bfill',inplace=True)
-
-    location = file.split('-')[0]
-    temp_frame["Location"] = location
+def preprocessing():
+
+    #########  Now dealing with 2003 to 2012 ############
+    fileNames = ['Alamosa-2003-2012.csv','Denver-2003-2012.csv','Grand-2003-2012.csv','Springs-2003-2012.csv']
+    path = "C:/Users/ramosv/Desktop/GitHub/Weather-Prediction-Model/raw_data/Weather Prediction Raw Data/"
+    allFrames = []
+
+    for file in fileNames:
+        #os.path.join(path,file)...
+        temp_frame = pd.read_csv(path+file,usecols=["DATE","HourlyDryBulbTemperature"])

```

Comment:
This could be considered an "unsafe" path. It works for you system, but not mine. Relative path (and creation of system-independent absolute paths) is tricky but definitely worth figuring out for collaboration.

```suggestion
        temp_frame = pd.read_csv(os.path.join(path, file),usecols=["DATE","HourlyDryBulbTemperature"])

```

https://stackoverflow.com/questions/7132861/how-can-i-create-a-full-path-to-a-file-from-parts-e-g-path-to-the-folder-name

Reply:
Tested it and it works